lib/connect.c

```diff
@@ -1516,6 +1516,11 @@ CURLcode Curl_socket(struct connectdata *conn,
     /* no socket, no connection */
     return CURLE_COULDNT_CONNECT;
 
+  if(conn->transport == TRNSPRT_QUIC) {
```

Comment:
Is this the right check for deciding whether to set the socket to nonblocking mode?